### PR TITLE
Implemented Users Model

### PR DESCRIPTION
**Pull request by ChrisKheng**
* A test version of the users model.
@team-mariners

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class User < ApplicationRecord
+    # Enforces unique not null constraint on username and email
+    validates :username, presence: true,
+              uniqueness: { case_sensitive: false },
+              length: {minimum: 3, maximum: 25}
+
+    VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
```
Comment (dcshzj) [inline app/models/user.rb]: Is the `\A` and `\z` part of the regex? It seems like it is going to match the literal "A" and "z", which is probably not intended.

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    validates :email, presence: true,
+              uniqueness: { case_sensitive: false },
+              length: { maximum: 105 },
+              format: { with: VALID_EMAIL_REGEX }
+
+    has_secure_password
+end
```
Comment (dcshzj) [inline app/models/user.rb]: Can I check if we are also creating the `uid` column, or are we leaving it to rails to generate it as part of the table?

File: db/migrate/20200214150738_add_password_digest_to_users.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddPasswordDigestToUsers < ActiveRecord::Migration[6.0]
+  def change
+    add_column :users, :password_digest, :string
```
Comment (dcshzj) [inline db/migrate/20200214150738_add_password_digest_to_users.rb]: Can this migration be merged with the migration for create users table?

File: app/models/user.rb (hunk 2 above)
Comment (ChrisKheng) [reply app/models/user.rb]: I think we just leave it to rails to generate it as part of the table.

File: app/models/user.rb (hunk 1 above)
Comment (ChrisKheng) [reply app/models/user.rb]: Yep it's part of the regex, \A matches the start of the string, \z matches end of the string.

File: db/migrate/20200214150738_add_password_digest_to_users.rb (hunk 3 above)
Comment (ChrisKheng) [reply db/migrate/20200214150738_add_password_digest_to_users.rb]: Uhm I'm not sure about this

**Pull request closed without merging**